### ISSUE TITLE
belongs_to with custom foreign_key is breaking in 2.0.0.rc

### ISSUE THREAD

**Issue by maokomioko** [sferik/rails_admin#3184]
Having a model with indirect association:
`belongs_to :organization, primary_key: :org_number, foreign_key: :org_number`

Belongs to association dropdowns return object's ID (as primary key) instead of customly specified key.

This looks like a regression from here: 
https://github.com/sferik/rails_admin/commit/be7d2f4a3ad1fda788f92a0e0dd4a83b98f141f4#diff-3ccbdf28331673e0d29522ba324bddcfR81

![image](https://user-images.githubusercontent.com/767496/62798732-4cc59000-badf-11e9-9389-90ad2e9507e4.png)

Just to be sure I switched back to the earlier commit and dropdown values are filled with org_number as expected.

**Comment by mshibuya** [comment]
> Belongs to association dropdowns return object's ID (as primary key) instead of customly specified key.

Yeah that's the intention of my fix. What are you trying to do? How does your schema look like and why do you need previous behavior?

**Comment by maokomioko** [comment]
Here is a more detailed look.
```
class EditorOrganizationRole < OrganizationRole
end

class OrganizationRole < ApplicationRecord
  belongs_to :organization, primary_key: :org_number, foreign_key: :org_number
end

class Organization < ApplicationRecord
end
```

```
create_table "organizations". force: :cascade do |t|
  t.string "org_number"
  # other fields
end
```

I'm trying to either create or modify EditorOrganizationRole. 

A dropdown for organizations in production (pre-patch version) returns the org_number as a key:
![image](https://user-images.githubusercontent.com/767496/62820269-37de1080-bb62-11e9-8736-c0e3a01710f9.png)

The same dropdown in the current version (after-patch) returns an ID instead:
![image](https://user-images.githubusercontent.com/767496/62820281-57753900-bb62-11e9-9ea1-aedb80fd1050.png)

And that leads to the validation error, because editor_organization_role#org_number is getting organization's ID and lookup fails.

![image](https://user-images.githubusercontent.com/767496/62820294-88556e00-bb62-11e9-8f74-1c0f739aacdb.png)

**Comment by mshibuya** [comment]
Sorry it was my fault, just pushed a fix 0e92ca43 in the master.
Thanks for reporting!
